contributors/design-proposals/storage/container-storage-interface-cluster-registry.md

```diff
@@ -152,7 +142,15 @@ The driver may set any optional configuration fields (like `SkipAttach`) as appr
 When the driver is ready to serve, it must set `Ready` in the status to `true`.
 
 #### Upgrade/Downgrade
-TODO
+This change is backwards compatible.
+Existing CSI drivers that are already deployed will not create a `CSIDriver` object.
+And Kubernetes will continue to interact with those drivers as it does today.
+New drivers MAY create a `CSIDriver` object to change how Kubernetes interacts with them.
+
+If a Kubernetes cluster is downgraded to a version that does not support `CSIDriver`, but continues to have a CSI driver deployed on it that creates a `CSIDriver` object and expects non-default behavior, Kubernetes will not be able to interact with it correctly (e.g. it may call attach when driver requests attach not to be called resulting in the volumes not mounting). Therefore, cluster admins must ensure if they downgrade, that they also change the CSI driver, if needed.
```

Comment:
It's very likely that the driver will be the same in this case. If I have Gluster CSI driver working in 1.11 with external attacher, I can use the same driver in 1.12 with the external attacher or I may create CSIDriver for it and stop the attacher. The driver itself is the same in all three cases, PVs are the same and pod definitions are the same.

Downgrade from a non-attachable driver with a CSIDriver instance to 1.11 is not that easy as you describe, some volumes can be mounted to running pods and we need to make sure they remain mounted after downgrade and unmounted when the pod is deleted. Let's discuss in #2523